src/components/Disputes/reducer.js

```diff
@@ -0,0 +1,26 @@
+import * as DisputesTypes from './types'
+
+export const reduceDispute = dispute => {
+  return {
+    ...dispute,
+    createdAt: parseInt(dispute.createdAt, 10) * 1000,
+    state: DisputesTypes.convertFromString(dispute.state),
+    reducedState:
+      dispute.state === DisputesTypes.Phase.Ruled
+        ? DisputesTypes.Status.Closed
+        : DisputesTypes.Status.Open,
+
+    /* TO-DO  SUPER HORRIBE FOR NOW - This should be reduced using Facu's suggestions about how to calculate it */
```

Comment:
Do we want to update the logic or the coding style? For the style, maybe we could make it a bit more readable by having a function that returns early:

```jsx
function getCurrentPhase(dispute) {
  const { convertFromString, Phase } = DisputesTypes

  if (convertFromString(dispute.state) === Phase.JuryDrafting) {
    return Phase.JuryDrafting
  }

  if (convertFromString(dispute.state) === Phase.Evidence) {
    return Phase.Evidence
  }

  if (dispute.rounds && dispute.rounds.length > 0) {
    return convertFromString(
      dispute.rounds[dispute.rounds.length - 1].state
    )
  }

  return Phase.Invalid
}
```

Reply:
That was totally changed on the dispute-detail PR [see here](https://github.com/aragon/court-dashboard/blob/1e42271a9fa9dd08b3bfce9e36a6d473b283dfeb/src/hooks/useDisputes.js#L14)